parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -33,10 +33,12 @@
 
 abstract public class Binary implements Comparable<Binary>, Serializable {
 
+  private boolean isUnmodifiable;
```

Comment:
I think it would be easier to reason about this if this were named `isReused`. That immediately signals that the content might change. `isUnmodifiable` signals that we shouldn't do things that modify the content, but that's not really correct because it might be the caller that has wrapped a byte array that changes the content.